src/main/java/org/cornutum/regexpgen/js/Provider.java

```diff
@@ -31,7 +32,16 @@ public static org.cornutum.regexpgen.Provider forEcmaScript()
    */
   public RegExpGen matching( String regexp)
     {
-    return new Parser( regexp).parse( false); 
+    return matching( regexp, null);
```

Comment:
This original form of `matching` should be moved to the top-level Provider interface as a `default` method.